elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -441,6 +448,27 @@ public boolean updateAttribute(String fieldName, Object newVal) {
         return false;
     }
 
+    private void updateComplexAttribute(EntityDictionary dictionary,
+                                        String fieldName,
+                                        Object update,
+                                        Object original,
+                                        RequestScope scope) {
+        final Map<String, Object> updateToMap = TypeHelper.toMap(update);
```

Comment:
You can assume update will be Map or you can just call dictionary.setValueChecked.  You don't need to convert it.

Reply:
There is confusion here and I need you help further clarify:
1. There is no such method `dictionary.setValueChecked`.
2. I have to convert the `update` object to `Map` since we don't want to lose information about which fields client actually sent along in request [as you recommended earlier](https://github.com/yahoo/elide/pull/2283/files#r698722236)
I cannot cast the `update` object to `Map` either due to `ClassCastException`.

Reply:
1. look in PersistentResource - not dictionary.
2. update _is_ a Map.  Just check its type (instanceof).  If it's not a map, you can fallback to just setting the value directly.